main.go

```diff
@@ -84,8 +85,31 @@ func cliLinterOverrides(app *kingpin.Application, element *kingpin.ParseElement,
 	return nil
 }
 
+func loadDefaultConfig(app *kingpin.Application, element *kingpin.ParseElement, ctx *kingpin.ParseContext) error {
+	if element != nil {
+		return nil
+	}
+
+	for _, elem := range ctx.Elements {
+		if elem.OneOf.Flag == app.GetFlag("config") {
+			return nil
+		}
+	}
+
+	rcfile, found, err := findRCFile()
+	if err != nil || !found {
+		return err
+	}
+
+	return loadConfigFile(rcfile)
+}
+
 func loadConfig(app *kingpin.Application, element *kingpin.ParseElement, ctx *kingpin.ParseContext) error {
-	r, err := os.Open(*element.Value)
+	return loadConfigFile(*element.Value)
+}
+
+func loadConfigFile(filename string) error {
```

Comment:
It would be great to move this into `config.go` now. We've been trying to split up `main.go`. Things that use `kingpin` probably still belong in `main.go`, but anything config specific could go into `config.go` (loadConfigFile, findRCFile, findRCFileInDir), and the test cases for them can move into `config_test.go`